### PR TITLE
 Failure to delete the only element of `Vec` of length 1 

### LINKED ISSUE #4444: Failure to delete the only element of `Vec` of length 1
When one has a vector of length 1, removing it's on;y element reverts.

### PR DESCRIPTION
**Pull request by rostyslavtyshko**
## Description

Closes #4444 

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by mohammadfawaz — APPROVED** [review]
Good catch. Thanks.

File: sway-lib-std/src/vec.sw [hunk 1]
```diff
@@ -447,3 +449,10 @@ impl<T> From<raw_slice> for Vec<T> {
         asm(ptr: (self.buf.ptr(), self.len)) { ptr: raw_slice }
     }
 }
+
+#[test()]
+fn test_vec_with_len_1() {
```
Comment (Braqzen) [inline sway-lib-std/src/vec.sw:454]: Can we have some assertions (pre-post conditions) to turn this into a test? Atm this isn't technically a test

**Review by Braqzen — COMMENTED** [review]
Hm, I'm interested in knowing how this made it in since I thought that I have reviewed the vec a few times and I would have pointed out pre-post conditions

File: sway-lib-std/src/vec.sw (hunk 1 above)
Comment (rostyslavtyshko) [reply sway-lib-std/src/vec.sw:454]: Added some here https://github.com/FuelLabs/sway/pull/4445/commits/8df9b1fdfeb20ae26dc411a3218c5aa10f8234ac

**Review by Braqzen — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**